pep-0517.txt

```diff
@@ -786,28 +787,32 @@ any change can go live, and that any changes will necessarily be
 restricted to new releases.
 
 One specific consequence of this is that in this PEP, we're able to
-make the ``prepare_wheel_metadata`` command optional. In our design, this
-can easily be worked around by a tool like ``pip``, which can put code
-in its subprocess runner like::
-
-    def prepare_wheel_metadata(output_dir, config_settings):
-         if hasattr(backend, "prepare_wheel_metadata"):
-             backend.prepare_wheel_metadata(output_dir, config_settings)
-         else:
-             backend.build_wheel(output_dir, config_settings)
-             touch(output_dir / "PIP_ALREADY_BUILT_WHEELS")
-             unzip_metadata(output_dir/*.whl)
-
-    def build_wheel(output_dir, config_settings, metadata_dir):
-         if os.path.exists(metadata_dir.parent / "PIP_ALREADY_BUILT_WHEELS"):
-             copy(metadata_dir.parent / *.whl, output_dir)
-         else:
-             backend.build_wheel(output_dir, config_settings, metadata_dir)
-
-and thus expose a totally uniform interface to the rest of ``pip``,
+make the ``prepare_metadata_for_build_wheel`` command optional. In our design,
+this can be readily handled by build frontends, which can put code in
+their subprocess runner like::
+
+    def dump_wheel_metadata(backend, output_dir, config_settings):
+        if hasattr(backend, "prepare_metadata_for_build_wheel"):
+            backend.prepare_metadata_for_build_wheel(output_dir, config_settings)
+        else:
+            wheel_fname = backend.build_wheel(output_dir, config_settings)
+            (output_dir / "ALREADY_BUILT_WHEEL").write_text(wheel_fname)
+            unzip_metadata(output_dir / wheel_fname)
+
+    def ensure_wheel_is_built(backend, output_dir, config_settings, metadata_dir):
+        already_built = metadata_dir.parent / "ALREADY_BUILT_WHEEL"
+        if already_built.exists():
+            wheel_fname = already_built.read_text()
+            copy(metadata_dir.parent / wheel_fname, output_dir)
+        else:
+            backend.build_wheel(output_dir, config_settings, metadata_dir)
```

Comment:
The functions you're defining in this example are not hook implementations, but they still pass Path objects to the hooks, which is not compliant with the current spec as I understand it. You could `str()` them when you call the hooks, but I think it would be clearer to consistently use string paths in examples.